gemini/use-cases/applying-llms-to-data/gemini-and-documentai-for-entity-extraction/entity_processor.py

```diff
@@ -0,0 +1,48 @@
+import json
+import mimetypes
+from typing import Dict
+
+from google.cloud import documentai
+from vertexai.generative_models import GenerativeModel, Part
+
+
+class EntityExtractor:
+    """Abstract class representing an entity"""
+
+    def extract_entities(self) -> Dict:
+        """Abstract function"""
+        raise NotImplementedError()
```

Comment:
Use an Abstract Base Class for this

https://www.geeksforgeeks.org/abstract-classes-in-python/

Reply:
we added ABC class